src/python/pants/backend/go/sdk.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2021 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+import shlex
+import textwrap
+from dataclasses import dataclass
+from typing import Optional, Tuple
+
+from pants.backend.go.distribution import GoLangDistribution
+from pants.core.util_rules.external_tool import DownloadedExternalTool, ExternalToolRequest
+from pants.engine.fs import CreateDigest, Digest, FileContent, MergeDigests
+from pants.engine.internals.selectors import Get
+from pants.engine.platform import Platform
+from pants.engine.process import BashBinary, Process
+from pants.engine.rules import collect_rules, rule
+from pants.util.logging import LogLevel
+
+
+@dataclass(frozen=True)
+class SetupGoSdkProcess:
+    digest: Digest
```

Comment:
This should probably be optional w/ default `EMPTY_DIGEST`. Maybe rename to `input_digest` to make abundantly clear what this is.

Reply:
Renamed but kept as required. Most invocations (other than the SDK import path analysis) will pass this so better to keep it required for now.